### PR TITLE
[WIP] Track retry group status

### PR DESCRIPTION
**Pull request by WilliamBZA**
Superceded by https://github.com/Particular/ServiceControl/pull/862

~~Work in progress. Submitting PR to begin collaboration early.

Connects to https://github.com/Particular/PlatformDevelopment/issues/936

Introduces a concept of a `RetryOperation` to represent the entire retry operation.  
1. `FailureGroupsRetryApi` receives a request to retry a group
2. The in-memory status for that group is set to `Preparing`
3. A `RetryAllInGroup` message is sent on the SC bus
4. The `RetryAllInGroupHandler` checks if there is an existing `RetryOperation` in progress for that group (_note: this may be a mistake in the case of zombie `RetryOperation`'s_)
5. A retry request is en queued onto the in-memory concurrent queue
6. When a timer triggers, the retry request is dequeued, the messages within the group are split into batches
7. A `RetryOperation` request is inserted into Raven for the requested operation
8. Each `RetryBatch` for the operation is inserted into the DB and all the corresponding messages are inserted into the `FailedMessageRetries` index
9. The `RetryProcessor` picks up that the batches are marked and are ready to be staged.
10. Once each batch is picked up, it is sent to the forwarding state
11. Once a batch has been forwarded, the number of remaining batches on the `RetryOperation` document for that batch is decreased by 1
12. If there are no more batches remaining, the `RetryOperation` document is deleted from the database
~~

### CODE REVIEW

File: src/ServiceControlInstaller.CustomActions.UnitTests/ServiceControlInstaller.CustomActions.UnitTests.csproj [hunk 1]
```diff
@@ -75,6 +75,9 @@
       <Name>ServiceControlInstaller.CustomActions</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (johnsimons) [inline src/ServiceControlInstaller.CustomActions.UnitTests/ServiceControlInstaller.CustomActions.UnitTests.csproj]: I am ok with keeping it

**Comment by gbiellem** [comment]
@WilliamBZA   If you rebase the App_Packages inspection error should disappear

**Comment by janovesk** [comment]
Brought in the map-reduce index for rebuilding retry operation status on startup.

**Comment by janovesk** [comment]
Superceded by https://github.com/Particular/ServiceControl/pull/862

**Comment by WilliamBZA** [comment]
bye

**Pull request closed without merging**